tifftools/commands.py

```diff
@@ -572,6 +584,163 @@ def tiff_set(source, output=None, overwrite=False, setlist=None, unset=None,
+                        2 if geotype == 'geographic_2d' else 1
+                    ),
+                    GTRasterType=1,
+                    GeographicType=projection,
+                )
+
+    if geokeys is None:
+        try:
+            import pyproj
+        except ImportError:
+            msg = (
+                f'pyproj is required to interpret projection {projection}. '
+                'Please install tifftools[geo].'
+            )
+            raise TifftoolsError(msg)
+
+        proj = pyproj.CRS.from_string(projection)
+        proj_dict = proj.to_dict()
+        angular_units = GeoTiffAngularUnits.get(proj.prime_meridian.unit_name).value
+        geokeys = dict(
+            GTModelType=3 if proj.is_geocentric else 2 if proj.is_geographic else 1,
+            GTRasterType=1,
+            GeographicType=proj.to_epsg(),
+            GeogCitation=proj_dict.get('datum'),
+            GeogAngularUnits=angular_units,
+            GeogSemiMajorAxis=proj.ellipsoid.semi_major_metre,
+            GeogInvFlattening=proj.ellipsoid.inverse_flattening,
+            ProjStdParallel1=proj_dict.get('lat_1'),
+            ProjStdParallel2=proj_dict.get('lat_2'),
+            ProjNatOriginLat=proj_dict.get('lat_0'),
+            ProjNatOriginLong=proj_dict.get('lon_0'),
+        )
+        if proj.coordinate_operation:
+            method_name = proj.coordinate_operation.method_name
+            transformation = GeoTiffTransformations.get(method_name.replace(' ', ''))
+            geokeys['GTCitation'] = method_name
+            geokeys['ProjCoordTrans'] = transformation
+        if proj_dict.get('units'):
+            linear_units = GeoTiffLinearUnits.get(proj_dict.get('units')).value
+            geokeys['ProjLinearUnits'] = linear_units
+
+    geokeys = {k: v for k, v in geokeys.items() if v is not None}
+
+    key_directory_version = 1
+    key_revision = 1
+    minor_revision = 0
+    number_of_keys = len(geokeys)
+    header = ' '.join(str(v) for v in [
+        key_directory_version,
+        key_revision,
+        minor_revision,
+        number_of_keys,
+    ])
+
+    geokey_tag = f'{header}'
+    doubles = []
+    asciis = []
+    for geokey, value in geokeys.items():
+        key_spec = GeoTiffGeoKey.get(geokey)
+        key_id = key_spec.value
+        data_type = key_spec.datatype
+        value_count = 1
+        if data_type == Datatype.SHORT:
+            tag_location = 0
+            value_offset = int(value)
+        elif data_type == Datatype.DOUBLE:
+            tag_location = Tag.get('GeoDoubleParamsTag').value
+            value_offset = len(doubles)
+            doubles.append(float(value))
+        else:
+            tag_location = Tag.get('GeoAsciiParamsTag').value
+            value_offset = len(asciis)
+            asciis.append(str(value))
+        geokey_tag += f' {key_id} {tag_location} {value_count} {value_offset}'
+    doubles_tag = ' '.join(str(v) for v in doubles)
```

Comment:
This is a clever way to leverage the existing functions.  At first I thought that maybe we should try to avoid this conversion to string and back to float later, but modern python is numerically stable, so this won't introduce any variation or error.